packages/flutter/test/cupertino/action_sheet_test.dart

```diff
@@ -0,0 +1,957 @@
+    // Set the scroll position back to zero.
+    messageScrollController.jumpTo(0.0);
+
+    // Expect the action sheet to take all available height.
+    expect(tester.getSize(find.byType(CupertinoActionSheet)).height, screenHeight);
+  });
+
+  testWidgets('Tap on button calls onPressed', (WidgetTester tester) async {
+    bool wasPressed = false;
+    await tester.pumpWidget(
+      createAppWithButtonThatLaunchesActionSheet(
+          new Builder(builder: (BuildContext context) {
+            return new CupertinoActionSheet(
+              actions: <Widget>[
+                CupertinoActionSheetAction(
+                  child: const Text('One'),
+                  onPressed: () {
+                    wasPressed = true;
+                    Navigator.pop(context);
+                  },
+                ),
+              ],
+            );
+          },
+        ),
+      ),
+    );
+
+    await tester.tap(find.text('Go'));
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+
+    expect(wasPressed, isFalse);
+
+    await tester.tap(find.text('One'));
+
+    expect(wasPressed, isTrue);
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+
+    expect(find.text('One'), findsNothing);
+  });
+
+  testWidgets('Action sheet width is correct when given infinite horizontal space',
+          (WidgetTester tester) async {
+    await tester.pumpWidget(
+      createAppWithButtonThatLaunchesActionSheet(
+        new Row(
+          children: <Widget>[
+            new CupertinoActionSheet(
+              actions: <Widget>[
+                CupertinoActionSheetAction(
+                  child: const Text('One'),
+                  onPressed: () {},
+                ),
+                CupertinoActionSheetAction(
+                  child: const Text('Two'),
+                  onPressed: () {},
+                ),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+
+    await tester.tap(find.text('Go'));
+    await tester.pump();
+
+    expect(tester.getSize(find.byType(CupertinoActionSheet)).width, 600.0);
+  });
+
+  testWidgets('Action sheet height is correct when given infinite vertical space',
+          (WidgetTester tester) async {
+    await tester.pumpWidget(
+      createAppWithButtonThatLaunchesActionSheet(
+        new Column(
+          children: <Widget>[
+            new CupertinoActionSheet(
+              actions: <Widget>[
+                CupertinoActionSheetAction(
+                  child: const Text('One'),
+                  onPressed: () {},
+                ),
+                CupertinoActionSheetAction(
+                  child: const Text('Two'),
+                  onPressed: () {},
+                ),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+
+    await tester.tap(find.text('Go'));
+    await tester.pump();
+
+    expect(tester.getSize(find.byType(CupertinoActionSheet)).height, 132.33333333333334);
```

Comment:
moreOrLessEquals (or it can be undeterministic depending on the machine it's running on)